# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLValidateStrings.h
# Commit: 7e2a2b1b416f
# Full Hash: 7e2a2b1b416f1a64c6548aab29e92acc9554ae7d
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Fix all webgl regression tests according to CI. r=handyman
#   
#   (This is a combination of 31 commits)
#   
#   * Fix Linux compilation.
# ==============================================================================

diff -r ccfa767dba64 -r 7e2a2b1b416f dom/canvas/WebGLValidateStrings.h
--- a/dom/canvas/WebGLValidateStrings.h	Wed Jan 08 22:19:16 2020 +0000
+++ b/dom/canvas/WebGLValidateStrings.h	Wed Jan 08 22:19:23 2020 +0000
@@ -27,15 +27,18 @@
 #ifndef WEBGL_VALIDATE_STRINGS_H_
 #define WEBGL_VALIDATE_STRINGS_H_
 
-#include "nsString.h"
+#include "WebGLTypes.h"
+
+#include "mozilla/Maybe.h"
+
+#include <string>
 
 namespace mozilla {
 
-class WebGLContext;
-
-bool TruncateComments(const nsAString& src, nsAString* const out);
-bool ValidateGLSLPreprocString(WebGLContext* webgl, const std::string& string);
-bool ValidateGLSLVariableName(const std::string& name, WebGLContext* webgl);
+std::string CommentsToSpaces(const std::string& src);
+Maybe<char> CheckGLSLPreprocString(bool webgl2, const std::string& string);
+Maybe<webgl::ErrorInfo> CheckGLSLVariableName(const bool webgl2,
+                                              const std::string& name);
 
 }  // namespace mozilla
 